src/components/prelude/auth/mod.rs

```diff
@@ -65,18 +93,69 @@ pub fn Auth(cx: Scope<Props>) -> Element {
                         "{l.create_account_desc}",
                     },
                     div { class: "m-bottom" },
-
-
-                    div {
-                        class: "display",
-                            rsx! {
-                                Icon {
-                                    icon: Shape::User,
-                                    size: 30,
+                    div{
+                        class: "photo-picker",                  
+                        if profile_picture_is_empty {
+                                rsx! {
+                                    Icon {
+                                        icon: Shape::User,
+                                        size: 30,
+                                    }
+                                }
+                            } else {
+                                rsx!{
+                                    img {
+                                        class: "profile_photo",
+                                        src: "{profile_picture_state}",
+                                        height: "100",
+                                        width: "100",
+                                    }
                                 }
                             }
-                    },
+                        Button {
+                            icon: Shape::Plus,
+                            on_pressed: move |_| {
+                                let path = match FileDialog::new().add_filter("image", &["jpg", "png", "jpeg", "svg"]).set_directory(".").pick_file() {
+                                    Some(path) => path,
+                                    None => return
+                                };
+                                let file = match std::fs::read(&path) {
+                                    Ok(image_vec) => image_vec,
+                                    Err(_) => vec![],
+                                };
+                                let filename = std::path::Path::new(&path)
+                                .file_name()
+                                .unwrap_or_else(|| std::ffi::OsStr::new(""))
```

Comment:
```suggestion
                                .map(|s| s.to_string_lossy().to_string())
                                .unwrap_or_default();
```

Reply:
Note this suggestion requires removing everything after the `upwrap_or_default()` so i would suggest applying this suggestion in the IDE.